app.py

```diff
@@ -140,24 +84,95 @@ def message(event, client):
     message_ts = event.get("ts")
     message_thread_ts = event.get("thread_ts")
 
-    # logs the message in the terminal, but NOT write it to a text file.
-    if text and text.lower() == "start":
-        return start_onboarding(user_id, channel_id, client)
-
-    logger.info("{} sent message: {}".format(user_id, text))
-
-    # if the message is sent as a threaded reply within ten minutes of the original message, 
-    # it'll be written into messages.txt as the title of documentation. Will fix later.
-    try:    
-        print(float(message_ts) - float(message_thread_ts))
-        if float(message_ts) - float(message_thread_ts) <= 600:
-            client.chat_postMessage(channel=channel_id, text="Got it.", thread_ts=message_thread_ts)
-            # initializes the file for the messages and puts cursor at the last line
-            file = open("messages.txt", "a+")
-            file.write("Title of Documentation: {}\n".format(text))
-            file.close()
-    except TypeError:
-        return
+    # sends the onboarding message if "start" is typed.
+    try:
+        if text.lower() == "start":
+            return start_onboarding(user_id, channel_id, client)
+    except AttributeError:

```

Comment:
In hindsight I should've done that immediately when I found the error - I even had to go back to replicate how I got the error.

New code & comments have been added to explain what happens in the exception. 

I have a follow-up question: IF the error is minor & the comments explain the error fully, does the exception need to do anything? Or is it just fine leaving the exception code as None?

Reply:
Thanks for updating! Now it's much more clear what the error is and why it's ok to except it.

It's ideal to do something with the error, because errors are usually appearing for some reason/whatever is throwing the error isn't returning what you expect it to return. In cases like this, a simple log message is fine, but I would avoid just silently ignoring an error if possible.

Also, if you can build your explanation into your code/with logging, then it will always be clear to the user/developers what's going on, but comments won't be visible to the user.

Reply:
great! that makes sense. thanks for the explanation!